crates/plugins/src/store.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::{anyhow, Context, Result};
+use flate2::read::GzDecoder;
+use std::{
+    fs::{self, File},
+    path::{Path, PathBuf},
+};
+use tar::Archive;
+
+use crate::manifest::PluginManifest;
+
+/// Directory where the manifests of installed plugins are stored.
+pub const PLUGIN_MANIFESTS_DIRECTORY_NAME: &str = "manifests";
+
+/// Houses utilities for getting the path to Spin plugin directories.
+pub struct PluginStore {
+    root: PathBuf,
+}
+
+impl PluginStore {
+    pub fn new(root: impl AsRef<Path>) -> Self {
+        Self {
+            root: root.as_ref().to_owned(),
+        }
+    }
+
+    pub fn default() -> Result<Self> {
+        let data_dir = match std::env::var("TEST_PLUGINS_DIRECTORY") {
+            Ok(test_dir) => PathBuf::from(test_dir),
+            Err(_) => dirs::data_local_dir()
+                .or_else(|| dirs::home_dir().map(|p| p.join(".spin")))
+                .ok_or_else(|| anyhow!("Unable to get local data directory or home directory"))?,
+        };
+        let plugins_dir = data_dir.join("spin").join("plugins");
+        Ok(Self::new(plugins_dir))
+    }
+
+    /// Gets the path to where Spin plugin are installed.
+    pub fn get_plugins_directory(&self) -> &Path {
+        &self.root
+    }
+
+    /// Get the path to the subdirectory of an installed plugin.
+    pub fn plugin_subdirectory_path(&self, plugin_name: &str) -> PathBuf {
+        self.root.join(plugin_name)
+    }
+
+    /// Get the path to the manifests directory which contains the plugin manifests
+    /// of all installed Spin plugins.
+    pub fn installed_manifests_directory(&self) -> PathBuf {
+        self.root.join(PLUGIN_MANIFESTS_DIRECTORY_NAME)
+    }
+
+    pub fn installed_manifest_path(&self, plugin_name: &str) -> PathBuf {
+        self.installed_manifests_directory()
+            .join(manifest_file_name(plugin_name))
+    }
+
+    pub fn installed_binary_path(&self, plugin_name: &str) -> PathBuf {
+        let mut binary = self.root.join(plugin_name).join(plugin_name);
+        if cfg!(target_os = "windows") {
+            binary.set_extension("exe");
+        }
+        binary
+    }
+
+    /// Returns the PluginManifest for an installed plugin with a given name.
+    /// Looks up and parses the JSON plugin manifest file into object form.
+    pub fn read_plugin_manifest(&self, plugin_name: &str) -> Result<PluginManifest> {
+        let manifest_path = self.installed_manifest_path(plugin_name);
+        log::info!("Reading plugin manifest from {}", manifest_path.display());
```

Comment:
As with `format!`, you can inline simple variables and `.display()` is redundant.

```suggestion
        log::info!("Reading plugin manifest from {manifest_path}");
```

Reply:
same as above ... cannot display `Path` this way unfortunately